src/pages/[platform]/build-a-backend/more-features/in-app-messaging/in-app-messaging-migration-guide/index.mdx

```diff
@@ -39,138 +39,545 @@ export function getStaticProps(context) {
   };
 }
 
-## Initialization and syncing messages
+This guide will help you migrate Amplify JavaScript v5's In-App Messaging APIs to the new v6's In-App Messaging APIs.
 
-As of v6 of Amplify JavaScript, you will now import the functional API’s directly from the `aws-amplify/in-app-messaging` path as shown below. 
+## Initialization
 
-Note: Red lines of code are v5 and Green lines are v6.
+As of v6 of Amplify, In-App Messaging must first be initialized by using the `initializeInAppMessaging` API exported from the `aws-amplify/in-app-messaging` path. In a typical application, this step should be performed immediately after Amplify is configured.
 
-```diff
-- import { Notifications } from 'aws-amplify';
+```ts
+import { Amplify } from 'aws-amplify';
+import amplifyconfig from './amplifyconfiguration.json';
+import { initializeInAppMessaging } from 'aws-amplify/in-app-messaging';
 
-+ import {
-+   initializeInAppMessaging,
-+   syncMessages,
-+   setConflictHandler
-+ } from 'aws-amplify/in-app-messaging';
+Amplify.configure(amplifyconfig);
+initializeInAppMessaging();
+```
 
-- const { InAppMessaging } = Notifications;
-+ initializeInAppMessaging();
+## InAppMessaging.syncMessages
 
-- InAppMessaging.syncMessages();
-+ syncMessages();
-- InAppMessaging.setConflictHandler(myConflictHandler);
-+ setConflictHandler(myConflictHandler);
+The `syncMessages` API in v6 has not changed in behavior; however, it is now exported from the `aws-amplify/in-app-messaging` path.
 
-```
+<BlockSwitcher>
+  <Block name="V6">
+    ```ts
+    import { syncMessages } from 'aws-amplify/in-app-messaging';
 
-## Dispatch event and clear messages
+    await syncMessages();
+    ```
 
-```diff
-- import { Notifications } from 'aws-amplify';
+  </Block>
+  <Block name="V5">
+    ```ts
+    import { Notifications } from 'aws-amplify';
+    const { InAppMessaging } = Notifications;
 
-+ import {
-+   dispatchEvent,
-+   initializeInAppMessaging,
-+   clearMessages,
-+ } from 'aws-amplify/in-app-messaging';
+    await InAppMessaging.syncMessages();
+    ```
 
-- const { InAppMessaging } = Notifications;
-+ initializeInAppMessaging();
+  </Block>
+</BlockSwitcher>
 
-const sendEvent = (eventName: string) => {
--   InAppMessaging.dispatchEvent({ name: eventName });
-+   dispatchEvent({ name: eventName });
-}
+## Displaying messages
 
-- await InAppMessaging.clearMessages();
-+ await clearMessages();
+The way in-app messages are displayed in v6 has not changed in behavior. You will continue to use the `record` API from Analytics or the `dispatchEvent` API from In-App Messaging. However, these APIs are now exported from the `aws-amplify/analytics` and `aws-amplify/in-app-messaging` paths, respectively.
 
-```
+### Analytics.record
```

Comment:
Let's change the order to lead with the dispatchEvent and then Analytics.record. Seems like more idiomatic that way. If I understand correctly, functionally, it'll lead to the same outcome?

Reply:
My understanding is that it is actually more idiomatic to use `record`. I.e. as a side effect of an analytics event customer is already interested in, an in-app messaging campaign can be tied to it